subscriber/podaac_access.py

```diff
@@ -89,20 +92,38 @@ def setup_earthdata_login_auth(endpoint):
     request.install_opener(opener)
 
 
+
+def get_token(url: str) -> str:
+    tokens = list_tokens(url)
+    if len(tokens) == 0 :
+        return create_token(url)
+    else:
+        return tokens[0]
+
 ###############################################################################
 # GET TOKEN FROM CMR
 ###############################################################################
-def get_token(url: str, client_id: str, endpoint: str) -> str:
+@tenacity.retry(wait=tenacity.wait_random_exponential(multiplier=1, max=60),
+                stop=tenacity.stop_after_attempt(3),
+                reraise=True,
+                retry=(tenacity.retry_if_result(lambda x: x == ''))
+                )
+def create_token(url: str) -> str:
     try:
         token: str = ''
-        username, _, password = netrc.netrc().authenticators(endpoint)
-        xml: str = """<?xml version='1.0' encoding='utf-8'?>
-        <token><username>{}</username><password>{}</password><client_id>{}</client_id>
-        <user_ip_address>{}</user_ip_address></token>""".format(username, password, client_id, IPAddr)  # noqa E501
-        headers: Dict = {'Content-Type': 'application/xml', 'Accept': 'application/json'}  # noqa E501
-        resp = requests.post(url, headers=headers, data=xml)
+        username, _, password = netrc.netrc().authenticators(edl)
+        headers: Dict = {'Accept': 'application/json'}  # noqa E501
+
+
+        resp = requests.post(url+"/token", headers=headers, auth=HTTPBasicAuth(username, password))
         response_content: Dict = json.loads(resp.content)
-        token = response_content['token']['id']
+        if "error" in response_content:
+            if response_content["error"] == "max_token_limit":
+                logging.error("Max tokens acquired from URS. Deleting existing tokens")
```

Comment:
It's not actually deleting any right? Just returning the first existing one

Reply:
Correct, if we attempt to create a token and max is the response, we simply "list" the tokens and use an existing one.